internal/util/search/search.go

```diff
@@ -153,7 +153,11 @@ func (sr *SearchReplace) matchAndReplace(node *yaml.Node, path string) error {
 		}
 
 		if sr.filePath != "" {
-			pathVal := fmt.Sprintf("%s: %s", strings.TrimPrefix(path, PathDelimiter), node.Value)
+			nodeVal, err := yaml.String(node)
+			if err != nil {
+				return err
+			}
+			pathVal := fmt.Sprintf("fieldPath: %s\nvalue: %s", strings.TrimPrefix(path, PathDelimiter), strings.TrimSpace(nodeVal))
```

Comment:
This seems to tightly couple the search implementation with the output format.

Reply:
I was thinking of returning structured results instead. `SearchResult` with the the output data and pass it to CmdSearch so that the cli will be printing it. Api will not be tied to printing format. WDYT?

Reply:
Yeah, I'm thinking the same thing. The challenge could be how we should define the `SearchResult` data structure. We could try to see if `ExecuteCmdOnPkgs` can be unaware of this, so different commands can have their own `SearchResult` implementation.

Reply:
Yes. The result struct should not be tied to ExecuteCmdOnPkgs. Will implement it and we can discuss on that.

Reply:
@mortent Started off with a simple struct which is good enough to hold the results. PTAL